### PR TITLE
Fix TKG OVA template upgrade mechanism in CSE

### PR DESCRIPTION
**Pull request by adambarreiro**
## Description

The CSE UI Plugin allows to perform a cluster upgrade when the chosen OVA template only differs in the Kubernetes version (or in other words, the TKG version **remains the same**).

Our code, however, only allows to perform upgrades when both TKG version and Kubernetes version are higher, which is wrong.

This PR fixes this, by allowing the TKG version to be equal as well.

## Testing

**Environment:** For instance, we can upload these two OVAs in the `tkg` catalog in VCD (they have both **the same TKG version** v2.4.0):

- ubuntu-2004-kube-v1.26.8+vmware.1-tkg.1-b8c57a6c8c98d227f74e7b1a9eef27st
- ubuntu-2004-kube-v1.27.5+vmware.1-tkg.1-0eb96d2f9f4f705ac87c40633d4b69st

**govcd_test_config.yaml**: Set `ovaName: "ubuntu-2004-kube-v1.26.8+vmware.1-tkg.1-b8c57a6c8c98d227f74e7b1a9eef27st"` (the lowest Kubernetes version)

Then run **Test_Cse** (it should take 45-150min)

Without the patch, the update is skipped and it outputs:
```
WARNING: CseKubernetesCluster.UpgradeCluster method not tested. It was skipped as there's no OVA to upgrade the cluster
```

With the patch, upgrade should be run and the warning does not appear.

- [x] Passed with CSE 4.2.0 and VCD 10.5.1.1

### CODE REVIEW

**Review by Didainius — APPROVED** [review]

File: govcd/cse.go [hunk 1]
```diff
@@ -233,8 +233,8 @@ func (cluster *CseKubernetesCluster) GetSupportedUpgrades(refreshOvas bool) ([]*
 		if err != nil {
 			continue // This means it's not a TKGm OVA, or it is not supported, so we skip it
 		}
-		// The OVA can be used if the TKG version is higher than the actual and the Kubernetes version is at most 1 minor higher.
-		if targetVersions.compareTkgVersion(cluster.TkgVersion.String()) == 1 && targetVersions.kubernetesVersionIsOneMinorHigher(cluster.KubernetesVersion.String()) {
+		// The OVA can be used if the TKG version is higher or equal than the actual and the Kubernetes version is at most 1 minor higher.
```
Comment (dataclouder) [inline govcd/cse.go:236]:
```suggestion
		// The OVA can be used if the TKG version is equal to the actual or higher and the Kubernetes version is at most 1 minor higher.
```
"equal than" doesn't feel right

**Review by dataclouder — APPROVED** [review]

File: .changes/v2.24.0/663-bug-fixes.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+* Fixed an issue that prevented CSE Kubernetes clusters to be upgraded to an OVA with higher Kubernetes version but same TKG version,
+  and to an OVA with a higher patch version of Kubernetes [GH-663] 
+* Fixed an issue that prevented CSE Kubernetes clusters to be upgraded to TKG v2.5.0 with Kubernetes v1.26.11 as it
+  performed an invalid upgrade of CoreDNS [GH-663]
```
Comment (dataclouder) [inline .changes/v2.24.0/663-bug-fixes.md:4]:
```suggestion
* Fixed an issue that prevented CSE Kubernetes clusters from being upgraded to an OVA with higher Kubernetes version but same TKG version,
  and to an OVA with a higher patch version of Kubernetes [GH-663] 
* Fixed an issue that prevented CSE Kubernetes clusters from being upgraded to TKG v2.5.0 with Kubernetes v1.26.11 as it
  performed an invalid upgrade of CoreDNS [GH-663] 
```

File: govcd/cse.go (hunk 1 above)
Comment (adambarreiro) [reply govcd/cse.go]: Fixed, thanks!

File: .changes/v2.24.0/663-bug-fixes.md (hunk 2 above)
Comment (adambarreiro) [reply .changes/v2.24.0/663-bug-fixes.md]: Fixed, thanks!

**Pull request merged**